tests/test_models.py

```diff
@@ -302,6 +304,80 @@ def test_null_default(self):
         instance = NullMoneyFieldModel.objects.create()
         assert instance.field is None
 
+    def test_implicit_currency_field_not_nullable_when_money_field_not_nullable(self):
+        instance = ModelWithVanillaMoneyField(money=10, money_currency=None)
+        with pytest.raises(CurrencyDoesNotExist, match=r"No currency with code NONE"):
```

Comment:
I think it would be better to avoid implicit conversion to string and provide a more specific error message for `None`.

Reply:
I think I see what you're getting at, if my 5 cents are worth anything, I'm of the opinion that's within the domain of `py-moneyed`(?). These are the lines where `"NONE”` would originate from:

https://github.com/py-moneyed/py-moneyed/blob/c495642244f629e4ed330b25438edbd517175146/src/moneyed/classes.py#L193-L197

`None` would get cast to `"NONE"` before it's validated as a currency code.

In addition to that, if we look at the `__init__` signature of `class Money` in `py-moneyed` (that runs the linked code above) it doesn't define any support for passing `currency=None`;

https://github.com/py-moneyed/py-moneyed/blob/c495642244f629e4ed330b25438edbd517175146/src/moneyed/classes.py#L173-L179

https://github.com/py-moneyed/py-moneyed/blob/c495642244f629e4ed330b25438edbd517175146/src/moneyed/classes.py#L186-L189

As such, I'm leaning towards the interpretation that `None` _is already an_ unsupported value for a currency code, therefore resulting in undefined behaviour (although kindly enough with the version of `py-moneyed` that's used we get an error with `"NONE"` attempted as currency code).

---

I'm aware of that I linked from `HEAD` at `py-moneyed`, but even if we're using an older version of `py-moneyed` which defines this differently. I kind of don't think it's worth to duplicate the code to handle `None`:

https://github.com/py-moneyed/py-moneyed/blob/c495642244f629e4ed330b25438edbd517175146/src/moneyed/classes.py#L186-L189

But _if_ we should do something different for `None`, I think we should "backport" those lines, throwing a `TypeError`.

Reply:
I also can't remember why I added this test in the first place. Might have been just for coverage or so..

Reply:
Isn't the AssertionError raised in L221 what would be the expected outcome of this test case?

Reply:
Not really, since that line is for initiating a `MoneyField` on a model(default value for currency), while this is setting a value for an already initiated `MoneyField`.

Reply:
> I think it would be better to avoid implicit conversion to string and provide a more specific error message for `None`.

I don't have too strong opinions handling `None` in django-money and raising a `TypeError`, if that's what you guys think is best.

Let me know and I'll bump the PR.

Reply:
I'm happy that wrong input (perhaps an edge case) is blocked in a nice way that will guide the developer to solve the issue, so I think we can move on :+1:

Reply:
I changed it to raise a `TypeError` just now.

Reply:
@flaeppe - thanks for updating the solution!

@Stranger6667 - to me, this looks good and fine, does the elaboration and the update work for you?

Reply:
Sounds good to me!

Reply:
Thanks for all the details :)